src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/DefragWatchdog.cs

```diff
@@ -272,85 +387,75 @@ private double GetNumberParameterById(string id)
             return (double)value;
         }
 
-        private int GetThreads()
+        private async Task<int> GetThreads(CancellationToken cancellationToken)
         {
-            var value = GetNumberParameterById(ThreadsId);
+            var value = await GetNumberParameterById(ThreadsId, cancellationToken);
             return (int)value;
         }
 
-        private int GetHeartbeatPeriod()
+        private async Task<int> GetHeartbeatPeriod(CancellationToken cancellationToken)
         {
-            var value = GetNumberParameterById(HeartbeatPeriodSecId);
+            var value = await GetNumberParameterById(HeartbeatPeriodSecId, cancellationToken);
             return (int)value;
         }
 
-        private int GetHeartbeatTimeout()
+        private async Task<int> GetHeartbeatTimeout(CancellationToken cancellationToken)
         {
-            var value = GetNumberParameterById(HeartbeatTimeoutSecId);
+            var value = await GetNumberParameterById(HeartbeatTimeoutSecId, cancellationToken);
             return (int)value;
         }
 
-        private double GetPeriod()
+        private async Task<double> GetPeriod(CancellationToken cancellationToken)
         {
-            var value = GetNumberParameterById(PeriodSecId);
+            var value = await GetNumberParameterById(PeriodHourId, cancellationToken);
             return (double)value;
         }
 
-        private bool IsEnabled()
+        private async Task<bool> IsEnabled(CancellationToken cancellationToken)
         {
-            var value = GetNumberParameterById(IsEnabledId);
+            var value = await GetNumberParameterById(IsEnabledId, cancellationToken);
             return value == 1;
         }
 
-        public static Task StartTask(Action action, bool longRunning = true)
+        private async Task InitParamsAsync()
         {
-            var task = new Task(action, longRunning ? TaskCreationOptions.LongRunning : TaskCreationOptions.None);
-            task.Start(TaskScheduler.Default);
-            return task;
-        }
+            await Policy.Handle<SqlException>(e => !e.Message.Contains("login failed", StringComparison.OrdinalIgnoreCase) &&
+                                                   !e.Message.Contains("dbo.Parameters", StringComparison.OrdinalIgnoreCase))
+                .WaitAndRetryAsync(
+                    3,
+                    count => TimeSpan.FromSeconds(2),
+                    (ex, count) => _logger.LogWarning(ex, "Failed to initialize parameters. Retrying..."))
+                .ExecuteAsync(async () =>
+                {
```

Comment:
Are these retries needed after you added storage init notification?

Reply:
I'm ok to remove. I was trying to replicate previous behavior here